### PR TITLE
Slab support in data store

### PR DESCRIPTION
**Pull request by bvanessen**
Added support for placing a slab of a sample's data into the data
store.  Then each partition of data (slab) is manged as if it were a
subsample and exchanges within a fixed offset of nodes.  Also added
support in the data store to record the sample ID and slab ID.

### CODE REVIEW

**Comment by naoyam** [comment]
I don't see any issue with the changes. Let me try running this to see if the accuracy is not impacted.

**Comment by bvanessen** [comment]
Great.  I am trying to build on rzansel right now and still moving data over.  Waiting for an allocation on lassen.

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470

File: src/data_readers/data_reader_hdf5.cpp [hunk 1]
```diff
@@ -86,13 +124,73 @@ namespace lbann {
...
+      H5Dclose(h_data);
+    }
+    H5Fclose(h_file);
+    return;
   }
 
   void hdf5_reader::load() {
     lbann_comm* l_comm = get_comm();
+    if(dc::get_rank_stride() != 1) {
+      LBANN_ERROR("HDF5 MPI-IO data reader requires DistConv rank stride = 1");
```
Comment (naoyam) [inline src/data_readers/data_reader_hdf5.cpp:184]: Is there any reason with this requirement? I've been testing it with `rank_stride` == `num_partitions`.

**Comment by naoyam** [comment]
This is no longer necessary as the commits are adopted in #93.

**Pull request closed without merging**